<?xml version="1.0"?>
<launch>
	<node name="foveation_ros" pkg="foveated_yolt" type="foveation_ros" respawn="false" output="screen">
		<param name="width"   value="640"/>
		<param name="height"  value="480"/>
		<param name="levels"  value="5"/>
		<param name="sigma_x" value="100"/>
		<param name="sigma_y" value="100"/>
		<remap from="input_image" to="/usb_cam/image_raw"/>
		<remap from="output_image" to="foveated_image"/>
	</node>

	<node name="yolt_ros" pkg="foveated_yolt" type="yolt_ros" respawn="false" output="screen">
		<param name="width"   value="227"/>
		<param name="height"  value="227"/>
		<!--<param name="model_file" value="$(find foveated_yolt)/files/deploy_googlenet.prototxt"/>
		<param name="weight_file" value="$(find foveated_yolt)/files/bvlc_googlenet.caffemodel"/>-->
		<param name="model_file" value="$(find foveated_yolt)/files/deploy_squeezenet.prototxt"/>
		<param name="weight_file" value="$(find foveated_yolt)/files/squeezenet_v1.1.caffemodel"/>
		<!--<param name="model_file" value="$(find foveated_yolt)/files/deploy_mobilenet.prototxt"/>
		<param name="weight_file" value="$(find foveated_yolt)/files/mobilenet_v2.caffemodel"/>-->

		<param name="mean_file" value="$(find foveated_yolt)/files/imagenet_mean.binaryproto"/>
		<param name="label_file" value="$(find foveated_yolt)/files/synset_words_change.txt"/>
		<param name="top_classes" value="1"/>
		<param name="device" value="GPU"/>
		<remap from="input_image" to="foveated_image"/>
		<remap from="output" to="saliency_map"/>
	</node>

	<node name="usb_cam" pkg="usb_cam" type="usb_cam_node"  respawn="false" output="screen">
		<param name="pixel_format" value="yuyv" />
	</node>

	<node name="original_view" pkg="image_view" type="image_view" respawn="false" output="screen">
		<remap from="image" to="/usb_cam/image_raw"/>
	</node>

	<node name="foveation_view" pkg="image_view" type="image_view" respawn="false" output="screen">
		<remap from="image" to="/foveated_image"/>
	</node>

	<node name="yolt_view" pkg="image_view" type="image_view" respawn="false" output="screen">
		<remap from="image" to="/saliency_map"/>
	</node>
</launch>

